nx/lib/nx/binary_backend.ex

```diff
@@ -1726,6 +1726,55 @@ defmodule Nx.BinaryBackend do
     from_binary(out, result_data)
   end
 
+  @impl true
+  def take_along_axis(
+        %T{type: output_type} = output,
+        %T{shape: t_shape, type: {_, t_size} = t_type} = tensor,
+        %T{shape: idx_shape, type: {_, idx_size} = idx_type} = indices,
+        axis
+      ) do
+    permute = fn t ->
+      permutation =
+        t
+        |> Nx.axes()
+        |> List.delete(axis)
+        |> List.insert_at(Nx.rank(t) - 1, axis)
+
+      inverse_permutation =
+        permutation
+        |> Enum.with_index()
+        |> Enum.sort_by(fn {x, _} -> x end)
+        |> Enum.map(fn {_, i} -> i end)
+
+      shape_list = Tuple.to_list(output.shape)
+      permuted_shape = permutation |> Enum.map(&Enum.at(shape_list, &1)) |> List.to_tuple()
+
+      {permuted_shape, inverse_permutation}
+    end
+
+    {permuted_shape, inverse_permutation} = permute.(tensor)
```

Comment:
Does this need to be an anonymous function if you only invoke it once? :D Or is it for code organization?

Reply:
It doesn't!

In the first iteration this would have been called more than once, so it made sense. Now I can just move it to the function body